susemanager-utils/testing/automation/update-obs-environment-cr-project.sh

```diff
@@ -1,15 +1,93 @@
 #!/bin/bash
 
-if [ $# -ne 1 ];then
-    echo "usage: $0 N"
+parent_project="systemsmanagement:Uyuni:Master"
+
+usage_and_exit() {
+    echo "usage: $0 N [test_project]"
     echo "where N is the environment"
-    echo "$0 will add new packages in systemsmanagement:Uyuni:Master to systemsmanagement:Uyuni:Master:TEST:N:CR"
+    echo "and project is the test_project, which by default is ${parent_project}:N:CR" 
+    echo "$0 will add new packages from ${parent_project} to test_project"
+    exit -1
+}
+
+update_project() {
+    tproject=${1}
+    pproject=${2}
+    rname=${3}
+    echo "Updating ${tproject} from ${pproject} with repo ${rname}"
+    osc ls ${tproject} > /dev/null
+    if [ ${?} -ne 0 ];then
+        echo "Project ${tproject} does not exists. Creating ..."
+        echo ${new_project_config} |\
+            sed -e "s/__TEST_PROJECT_NAME__/${tproject}/g" |\
+            sed -e "s/__PARENT_PROJECT_NAME__/${pproject}/g" |\
+            sed -e "s/__REPO_NAME__/${rname}/g" > ${OUT}
+       osc meta prj --file=${OUT} -m "Created by ${0}" ${tproject}
+    fi
+    for i in $(diff <( osc ls ${tproject} ) <( osc ls ${pproject} ) | grep ">" | cut -d" " -f2);do
+        echo "Found new package ${i} in ${pproject}. Copying the RPMs to ${tproject}"
+        osc aggregatepac ${pproject} ${i} ${tproject}
+    done
+}
+
+if [ $# -lt 1 ];then
+    usage_and_exit
+fi
+if [ $# -gt 2 ];then
+    usage_and_exit
+fi
+
+test_parent_project="${parent_project}:TEST:${1}:CR"
+
+if [ $# -eq 2 ];then
+    test_parent_project=${2}
+fi
+
+parent_repo_name="openSUSE_Leap_15.3"
```

Comment:
I'd place both variables right below the shebang and the comments, so if we ever need to change this, it's easy to quick and harder to break things.

Reply:
both? parent_repo_name and which other one?

Reply:
Both. In the end this will change to `15.4` at some point :-)

Reply:
done